File: 1-two-sum/main.go
func twoSum(nums []int, target int) []int {
    
    _map := make(map[int]int)
    for idx, num := range nums {
        if value, ok := _map[num]; ok {
            return []int{idx, value}
        } 
        
        _map[target-num] = idx
    }

    return []int{}
}

File: 1-two-sum/main.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        
        map<int, int> _map;

        for (int i = 0; i < nums.size(); i++) {
            auto it = _map.find(nums[i]);
            if (it != _map.end()) {
                return {it -> second, i};
            }

            _map[target - nums[i]] = i;
        }

        return {};
    }
};
